moto/awslambda/models.py

```diff
@@ -515,23 +636,31 @@ def send_message(self, function_name, message):
             ]
 
         }
-        self._functions[function_name].invoke(json.dumps(event), {}, {})
+        self._functions[function_name][-1].invoke(json.dumps(event), {}, {})
         pass
 
     def list_tags(self, resource):
         return self.get_function_by_arn(resource).tags
 
     def tag_resource(self, resource, tags):
-        self.get_function_by_arn(resource).tags.update(tags)
+        fn = self.get_function_by_arn(resource)
+        if fn:
```

Comment:
minor suggestion: I'd personally prefer a guard clause here that early-returns on `if not fn` so the rest of the method isn't as indented. Same as the following method.

Reply:
Ahh, now we get down to personal preference :wink:. But I agree, less indentation is nice, and its an easy fix